src/app/qgssnappingwidget.cpp

```diff
@@ -199,6 +199,14 @@ QgsSnappingWidget::QgsSnappingWidget( QgsProject *project, QgsMapCanvas *canvas,
   tracingMenu->addAction( widgetAction );
   mEnableTracingAction->setMenu( tracingMenu );
 
+  // vertex tool may select vertices from all layers
+  mEditVerticesAllLayersAction = new QAction( tr( "Edit vertices of all editable layers at once" ), this );
+  mEditVerticesAllLayersAction->setCheckable( true );
+  mEditVerticesAllLayersAction->setIcon( QIcon( QgsApplication::getThemeIcon( "/mIconEditVerticesAllLayers.svg" ) ) );
+  mEditVerticesAllLayersAction->setToolTip( tr( "The vertex editor tool allows to edit vertices from many layers at once. This option allows to restore previous behavior, ie only edit active layer. Useful with transaction groups and database trigger design interfering with multi layer editing" ) );
+  mEditVerticesAllLayersAction->setObjectName( QStringLiteral( "EditVerticesAllLayersAction" ) );
+  connect( mEditVerticesAllLayersAction, &QAction::toggled, this, &QgsSnappingWidget::enableEditVerticesOnAllLayers );
+
```

Comment:
I would much rather see this setting somehow connected to vertex tool action (e.g. have a dropdown menu with vertex tool action with this setting, possibly more settings like this in the future)